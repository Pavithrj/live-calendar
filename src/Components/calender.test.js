import { act, render, screen } from "@testing-library/react"
import Calender from "./calender";

test("renders learn react link", () => {
    const { asFragment } = render(<Calender />);
    expect(asFragment()).toMatchSnapshot();
});

describe("Calender Component", () => {
    it('should show current year', () => {
        render(<Calender />);
        const container = screen.getByTestId('years');
        const currentYear = new Date().toLocaleDateString().split('/')[2];
        expect(currentYear).toBe(container.textContent);
    });

    it('should show current month', () => {
        render(<Calender />);
        const container = screen.getByTestId('months');
        const currentMonth = new Date().toLocaleString('default', { month: 'long' });
        expect(currentMonth).toBe(container.textContent);
    });

    it('should show current day', () => {
        render(<Calender />);
        const container = screen.getByTestId('days');
        const currentDay = new Date().toLocaleDateString('default', { weekday: 'long' });
        expect(currentDay).toBe(container.textContent);
    });

    it('should show current date', () => {
        render(<Calender />);
        const container = screen.getByTestId('dates');
        const currentDate = new Date().toLocaleDateString().split('/')[0];
        expect(currentDate).toBe(container.textContent);
    });

    it('should show current time', () => {
        render(<Calender />);
        const container = screen.getByTestId('times');
        const currentTime = new Date().toLocaleTimeString();
        // expect(currentTime).toBe(container);
    });

})