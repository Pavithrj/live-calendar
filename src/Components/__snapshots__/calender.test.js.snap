// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="month"
      data-testid="months"
    >
      May
    </div>
    <div
      class="day"
      data-testid="days"
    >
      Monday
    </div>
    <div
      class="date"
      data-testid="dates"
    >
      1
    </div>
    <div
      class="year"
      data-testid="years"
    >
      2023
    </div>
    <div
      class="time"
      data-testid="times"
    >
      Current Time: 
    </div>
  </div>
</DocumentFragment>
`;
